### ISSUE TITLE
Map instance storage to /tmp if available + has enough space

### ISSUE THREAD

**Issue by clv** [boxfuse/boxfuse-issues#179]
While the M, C & R instance families are going EBS only, the others are not, especially not the I & D (storage optimized) instance families. 

For instance types where there's a disk that comes with it, please map /tmp to that disk (as long as there's more space on that disk than has been requested in the boxfuse config). 

Then please add support for the additional instance families as they then make sense. 

Thanks!

**Comment by axelfontaine** [comment]
The T, M, C, and R do indeed have an EBS-only future.

What is your use-case for I & D instances? I am just trying to understand it better as they are uncommon choices for the type of transient AMI Boxfuse creates.

**Comment by clv** [comment]
Right, that's a fair point. That said, while our current instances (pre boxfuse) do survive deploys, they already routinely scale in and out. 

On one site we're using instance store for caching of ~7 gb of data from which webpages are created on the fly. The data is stored on S3, but cached after the first fetch to avoid the 200ms S3 penalty. Ideally we'd do that to ~700gb of data, and have been considering using the I family for that (right now we're on C3). 

On another we use it to store images while they're being processed, also ends up being 1-10 gb or so over the lifetime of the instance. 

Broadly speaking there are several reasons to support instance store: 

1. It's [*a lot* faster than EBS](https://gist.github.com/ktheory/3c3616fca42a3716346b), especially for small (1-30 gb) volumes. [AWS docs](http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/EBSVolumeTypes.html) - when you run out of burst credits, you're hosed on a small volume. I didn't see anything about bofuse /tmp using provisioned IOPS, but that gets pricey fast and is also limited by the size of the volume. Debugging performance problems on an instance I can't SSH to, where it only happens after the burst credits have been exhausted (which might take hours) is the stuff of devops nightmares... 

2. It's there and it's included. The C3 spot instances are glorious - there's not much competition for them, so you get unreasonably high reliability super-cheap, and they have just enough disk for what we need (caching a couple of gigs of data). While current & future versions won't have it, they're still useful and Amazon keeps the old instances around for years (we were even running a C1 instance until not long ago). 

3. As a user I don't really want my deploy system to dictate what instance types I can use (might be irrational, but yeah, it's still a factor). 

I'm not familiar with the internals of boxfuse, so I guess if it is really difficult / has a lot of ripple effects, then I can see not doing it (but isn't it "just" `mount ...`?) . 

I guess if people are used to taking snapshots of /tmp, then maybe they'll get confused? Maybe having a /tmp=instance-store or similar would make it a clean & clear opt-in? 

Look, I realize it's an obvious suggestion and that you've already thought about it and considered it not worth it, so I'm guessing there's a non-trivial amount of effort involved in getting it working. We can probably work around it, but it's a capability we've been using for years, and it's one of the few things we'd lose going with boxfuse.

**Comment by axelfontaine** [comment]
Thanks for detailing your situation. I can certainly see how I or D instances make sense in your case. What we could do in a first phase would be to add I and D instances to the list of instances supported by Boxfuse? You can then run your images on these instances and have your app invoke format and mount on startup for each instance-store volume you need. Would this work for you?
